### PR TITLE
Fixing #129, #130, #131 and #140.

### PR DESCRIPTION
**Pull request by tushargupta51**
Fixes for #129 #130 #131 #140
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23discussion_r30282334%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23discussion_r30326056%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23issuecomment-102056831%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23discussion_r30552484%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23discussion_r30553571%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23issuecomment-102056831%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40RandalliLama%20I%20have%20added%20fix%20for%20one%20more%20issue%2C%20renaming%20%5C%22tests%5C%22%20folder%20to%20%5C%22test%5C%22%20folder.%20This%20increased%20the%20number%20of%20files%20but%20they%20are%20mostly%20just%20renaming.%22%2C%20%22created_at%22%3A%20%222015-05-14T14%3A42%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2084f1794407f00c5ee1bce6b3dce281a7ab235b32%20src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23discussion_r30282334%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Could%20you%20put%20the%20code%20below%20the%20label%3F%22%2C%20%22created_at%22%3A%20%222015-05-13T22%3A24%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%2C%20%7B%22body%22%3A%20%22Done.%22%2C%20%22created_at%22%3A%20%222015-05-14T14%3A41%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs%3AL21-46%22%7D%2C%20%22Pull%20ed6c8de383c6feaf45937de99ee87fe9c4a7522f%20src/Microsoft.IdentityModel.Protocol.Extensions/project.json%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148%23discussion_r30552484%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20are%20we%20removing%20%27beta4%27%3F%22%2C%20%22created_at%22%3A%20%222015-05-18T22%3A08%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%2C%20%7B%22body%22%3A%20%22Having%20beta4%20was%20causing%20the%20warnings%20regarding%20the%20versions%20when%20building%20the%20project.%20The%20fix%20was%20either%20to%20change%20the%20%20%20%20%20%5C%22version%5C%22%3A%20%5C%222.0.0-%2A%5C%22%20to%20%5C%22version%3A%202.0.0-beta-%2A%5C%22%20in%20project.json%20or%20remove%20the%20%5C%22beta%5C%22%20from%20the%20dependencies.%20This%20is%20what%20asp.net%20does%20too.%20The%20beta%20is%20added%20in%20the%20build%20scripts.%22%2C%20%22created_at%22%3A%20%222015-05-18T22%3A26%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/Microsoft.IdentityModel.Protocol.Extensions/project.json%3AL3-11%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 84f1794407f00c5ee1bce6b3dce281a7ab235b32 src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs 8'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148#discussion_r30282334'>File: src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs:L21-46</a></b>
- <a href='https://github.com/RandalliLama'></a> Could you put the code below the label?
- <a href='https://github.com/tushargupta51'></a> Done.
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148#issuecomment-102056831'>General Comment</a></b>
- <a href='https://github.com/tushargupta51'></a> @RandalliLama I have added fix for one more issue, renaming "tests" folder to "test" folder. This increased the number of files but they are mostly just renaming.
- [ ] <a href='#crh-comment-Pull ed6c8de383c6feaf45937de99ee87fe9c4a7522f src/Microsoft.IdentityModel.Protocol.Extensions/project.json 5'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148#discussion_r30552484'>File: src/Microsoft.IdentityModel.Protocol.Extensions/project.json:L3-11</a></b>
- <a href='https://github.com/brentschmaltz'></a> Why are we removing 'beta4'?
- <a href='https://github.com/tushargupta51'></a> Having beta4 was causing the warnings regarding the versions when building the project. The fix was either to change the     "version": "2.0.0-*" to "version: 2.0.0-beta-*" in project.json or remove the "beta" from the dependencies. This is what asp.net does too. The beta is added in the build scripts.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/148'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs [hunk 1]
```diff
@@ -21,6 +21,25 @@ static IdentityModelEventSource()
         /// </summary>
         public static IdentityModelEventSource Logger { get; }
 
+        public void Write(EventLevel level, string message)
+        {
+            switch (level)
+            {
+                case EventLevel.Critical: WriteCritical(message);
```
Comment (RandalliLama) [inline src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs]: Could you put the code below the label?

Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs]: Done.

**Comment by tushargupta51** [comment]
@RandalliLama I have added fix for one more issue, renaming "tests" folder to "test" folder. This increased the number of files but they are mostly just renaming.

File: src/Microsoft.IdentityModel.Protocol.Extensions/project.json [hunk 2]
```diff
@@ -3,8 +3,8 @@
     "description": "Provides types that provide support for OpenIdConnect protocol.",
     "dependencies": {
         "Newtonsoft.Json": "6.0.6",
-        "System.IdentityModel.Tokens": "5.0.0-beta4-*",
-        "Microsoft.IdentityModel.Logging":  "1.0.0-beta4-*"
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocol.Extensions/project.json]: Why are we removing 'beta4'?

Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocol.Extensions/project.json]: Having beta4 was causing the warnings regarding the versions when building the project. The fix was either to change the     "version": "2.0.0-*" to "version: 2.0.0-beta-*" in project.json or remove the "beta" from the dependencies. This is what asp.net does too. The beta is added in the build scripts.

**Pull request merged**